### ISSUE TITLE
Training script stuck on DDP initialization

### ISSUE THREAD

**Issue by joshmyersdean** [mlfoundations/open_clip#631]
Hello! 

Thank you for this great work. I recognize this problem may not be unique to this repo but when training with multiple GPUS, the training script gets stuck on `model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[device], **ddp_args)`. I am using 3 A100s with the command:
```bash
torchrun --nproc-per-node 3 src/training/main.py  --train-data '{MY CSV FILE}'
```

Any advice would be greatly appreciated! I have set the master port and address as well as my world size to be 3.

**Comment by rwightman** [comment]
@joshmyersdean you have an issue with your hardware or pytorch/cuda/nccl versions. 

The command you quoted above will work for 3x GPU on the same machine if there are no issues with the software & hardware components. If it hangs you have problem unrelated to this repo. You might want to try explicitly masking GPUs in case you have > 3 on that machine (ie `CUDA_VISIBLE_DEVICES=0,1,2 torchrun --nproc-per-node 3 src/training/main.py  --train-data '{MY CSV FILE}`)

Doing this across machines requires setting ports and ips for master if you use torchrun, see extra rdvz / nnodes flags in the README example.

**Comment by rwightman** [comment]
Normally you don't need to be explicit with setting the port for single node case, unless you happen to be running more than one train setup on the same machine, then you need to make each one unique.

**Comment by joshmyersdean** [comment]
Thank you for the response, explicitly setting the devices didn't change anything so I will reach out to our IT folks to see what's going on.

Thanks again!
